File: atcoder/past201912_i.cpp
// https://atcoder.jp/contests/past201912-open/tasks/past201912_i
#include<iostream>
// #include<algorithm>
// #include<string>
// #include<numeric>
// #include<vector>
// #include<map>
// #include<tuple>
// #include<set>
// #include<queue>
// #include<deque>
// #include<regex>
// #include<bitset>
// #include<iomanip>
// #include<complex>
// #include<stack>
// #include<functional>

#include<bits/stdc++.h>
using namespace std;
#define rep(i,n) for (int i = 0; i < (n); ++i)
#define all(v) (v).begin(),(v).end()
// #define chmin(x,y) x = min(x,y)
// #define chmax(x,y) x = max(x,y)
template <class T> bool chmax(T &a, const T &b) { if (a < b) { a = b; return 1; } return 0; }
template <class T> bool chmin(T &a, const T &b) { if (b < a) { a = b; return 1; } return 0; }
template <class T> T gcd(T a, T b) { return b ? gcd(b, a % b) : a; }
template <class T> T lcm(T a, T b) { return a/gcd(a,b)*b; }
typedef pair<int, int> P;
typedef long long ll;
const int INF = 1001001001;
const ll LINF = 1001002003004005006ll;
const ll MOD = 1e9+7;
// ll gcd(ll a, ll b) { return b?gcd(b,a%b):a;}
// ll lcm(ll a, ll b) { return a/gcd(a,b)*b;}

int main() {
	int n, m;
	cin >> n >> m;
    vector<ll> dp(1<<n, LINF);
    dp[0] = 0;
    string s; int c;
    rep(i,m) {
        cin >> s >> c;
        ll k = 0;
        rep(i,n) if(s[i]=='Y') k |= (1 << (n-i-1));
        rep(j,(1<<n)) {
            chmin(dp[j | k], dp[j] + c);
        }
    }
    if (dp[(1<<n)-1] == LINF) {
        cout << -1 << endl;
    } else {
        cout << dp[(1<<n)-1] << endl;
    }
	return 0;
}

File: atcoder/past201912_i.py
# https://atcoder.jp/contests/past201912/tasks/past201912_i

INF = 1001001001

N, M = map(int, input().split())
sets = []
for i in range(M):
    s, c = input().split()
    b = int(s.replace('Y', '1').replace('N', '0'), 2)
    sets.append([b, int(c)])
# print(sets)

dp = [INF] * (1<<N)
dp[0] = 0
for b in range(1<<N):
    for i in range(M):
        t = b | sets[i][0]
        cost = dp[b] + sets[i][1]
        dp[t] = min(dp[t], cost)
ans = dp[-1]
if ans == INF:
    ans = -1
print(ans)

# key = []
# for i in range(M):
#     a, b = map(int, input().split())
#     s = 0
#     c = list(map(int, input().split()))
#     for j in range(b):
#         c[j] -= 1
#         s |= 1<<c[j]
#     key.append([s, a])
# dp = [INF] * (1<<N)
# dp[0] = 0
# for s in range(1<<N):
#     for i in range(M):
#         t = s | key[i][0]
#         cost = dp[s] + key[i][1]
#         dp[t] = min(dp[t], cost)
# ans = dp[-1]
# if ans == INF:
#     ans = -1
# print(ans)
